satpy/scene.py

```diff
@@ -803,11 +803,31 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         """Add the item to the scene."""
+        if key in self:
+            old_key = self._datasets.get_key(key)
+            self._wishlist.discard(old_key)
+            del self._datasets[old_key]
+
+        if isinstance(value, np.ndarray):
+            value = xr.DataArray(value)
```

Comment:
True maybe also checking for dask array is a good idea. If currently it is allowed to assign plain numpy arrays I guess assigning dask arrays is allowed to so they should also be converted to a `xr.DataArray`. I also thought about not only converting to a `xr.DataArray` if a user tries to assing a numpy array but also converting it to a dask array but can't remember why I didn't. I guess it makes sense. I will change it then.

Reply:
Forgot: I think non-DataArrays should not work at all. But I think there is no need for a warning because the conversion is kind of for the "benefit" of the user since many things might not work if the dataset is only a numpy array so the conversion, albeit intransparently, makes sure that only DataArrays are assinged.

Reply:
Ok so you're saying you think users should be able to provide a numpy array, or a dask array, or a DataArray (with numpy or dask underneath), and this will convert whatever is given to a DataArray with dask underneath?

If so, then I'm OK with that. If not, let's discuss.

Reply:
Yepp. I mean this would leave it backwards compatible if somebody is really providing numpy arrays in his scripts. I would also be ok with throwing a warning or exception and therefore really forbidding the assignment of numpy or dask arrays. That would be more transparent I guess. The question is if that is wanted. Basically the question is if we treat being able to provide numpy or dask arrays as a feature or not.

Reply:
I think I'm leaning towards not allowing non-DataArrays and if the DataArray has a numpy array underneath it should maybe be turned into a dask array. Here's what happens if you assign a numpy array to an xarray Dataset:


```
In [1]: import xarray as xr
In [2]: import numpy as np
In [3]: ds = xr.Dataset()
In [4]: ds["a"] = np.zeros((10, 10))
---------------------------------------------------------------------------
MissingDimensionsError                    Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 ds["a"] = np.zeros((10, 10))

...

MissingDimensionsError: cannot set variable 'a' with 2-dimensional data without explicit dimension names. Pass a tuple of (dims, data) instead.
```

Reply:
Ok then I will change that.